### DISCUSSION TITLE
Multiple Field Resolvers that call same API

### DISCUSSION THREAD

**Discussion by ckahn1** [MichalLytek/type-graphql#1535] [Q&A]
I have multiple field resolvers that interface with the same  REST API. If two of the fields that are annotated as "field resolver" are requested, that will initiate two identical API calls. To make this more efficient would it be possible to configure the field resolvers that interface with the same API to make just one request? I would like to avoid having to nest all of the fields into an object and make the object the field resolver.. 

```
person {
accountID, (calls api 1)
phoneNumber, -> field resolver (calls api 2)
email -> field resolver (calls api 2)
}
```

**Comment by MichalLytek** [comment]
Search for `dataloader` library or apollo RESTDataSource solution

**Reply by ckahn1** [reply]
@MichalLytek  can you use the RestDataSource in conjunction with field resolvers? Field resolvers execute in parallel correct?

**Comment by shivagithub1** [comment]
Hey @MichalLytek, 
Just curious on the solution you went with? I am also having similar problem and evaluating multiple options.
